srml/staking/src/lib.rs

```diff
@@ -1204,66 +1364,36 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	fn apply_force_new_era() {
-		ForceNewEra::put(true);
-	}
-
-	/// Call when a validator is determined to be offline. `count` is the
-	/// number of offenses the validator has committed.
+	/// Remove all associated data of a stash account from the staking system.
 	///
-	/// NOTE: This is called with the controller (not the stash) account id.
-	pub fn on_offline_validator(controller: T::AccountId, count: usize) {
-		let stash = if let Some(l) = Self::ledger(&controller) {
-			l.stash
-		} else {
-			return;
-		};
-
-		// Early exit if validator is invulnerable.
-		if Self::invulnerables().contains(&stash) {
-			return;
+	/// This is called :
+	/// - Immediately when an account's balance falls below existential deposit.
+	/// - after a `withdraw_unbond()` call that frees all of a stash's bonded balance.
```

Comment:
`withdraw_unbond` was removed in https://github.com/darwinia-network/darwinia/pull/92. Reserve?

Reply:
No need to reserve, if #92 is merged first, this PR will do the merging and resolve conflicts.